File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        int n = nums.length;
        
        if(n == 0)  return null;

        return getBST(nums, 0, n-1);
    }

    static TreeNode getBST(int nums[], int startInd, int endInd){
        if(startInd > endInd)   return null;

        int mid = startInd + (endInd - startInd) / 2;

        TreeNode root = new TreeNode(nums[mid]);

        root.left = getBST(nums, startInd, mid-1);
        root.right = getBST(nums, mid+1, endInd);

        return root;
    }
}

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: Optional[TreeNode]
        """
        def build_bst(lefty, righty):
            if lefty > righty:
                return None

            mid = (lefty + righty) // 2

            root = TreeNode(nums[mid])

            root.left = build_bst(lefty, mid-1)
            root.right = build_bst(mid+1, righty)

            return root

        return build_bst(0, len(nums)-1)

        